pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -65,72 +65,76 @@
  * The <code>InstancePlanMakerImplV2</code> class is the default implementation of {@link PlanMaker}.
  */
 public class InstancePlanMakerImplV2 implements PlanMaker {
-  // Instance config key for maximum number of threads used to execute the query
-  // Set as pinot.server.query.executor.max.execution.threads
-  public static final String MAX_EXECUTION_THREADS_KEY = "max.execution.threads";
-  public static final int DEFAULT_MAX_EXECUTION_THREADS = -1;
-
-  public static final String MAX_INITIAL_RESULT_HOLDER_CAPACITY_KEY = "max.init.group.holder.capacity";
-  public static final int DEFAULT_MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10_000;
-  public static final String MIN_INITIAL_INDEXED_TABLE_CAPACITY_KEY = "min.init.indexed.table.capacity";
-  public static final int DEFAULT_MIN_INITIAL_INDEXED_TABLE_CAPACITY = 128;
-  public static final String NUM_GROUPS_LIMIT_KEY = "num.groups.limit";
-  public static final int DEFAULT_NUM_GROUPS_LIMIT = 100_000;
-
-  // By default, group trimming in AggregateOperator is disabled
-  public static final int DEFAULT_GROUP_TRIM_SIZE = -1;
-
-  // Instance config key for minimum segment-level group trim size
-  // Set as pinot.server.query.executor.min.segment.group.trim.size
-  public static final String MIN_SEGMENT_GROUP_TRIM_SIZE_KEY = "min.segment.group.trim.size";
-  public static final int DEFAULT_MIN_SEGMENT_GROUP_TRIM_SIZE = -1;
-  // Instance config key for minimum server-level group trim size
-  // Caution: Setting it to non-positive value (disable trim) or large value can give more accurate result, but can
-  //          potentially cause memory issue
-  // Set as pinot.server.query.executor.min.server.group.trim.size
-  public static final String MIN_SERVER_GROUP_TRIM_SIZE_KEY = "min.server.group.trim.size";
-  public static final int DEFAULT_MIN_SERVER_GROUP_TRIM_SIZE = GroupByUtils.DEFAULT_MIN_NUM_GROUPS;
-  // set as pinot.server.query.executor.groupby.trim.threshold
-  public static final String GROUPBY_TRIM_THRESHOLD_KEY = "groupby.trim.threshold";
-  public static final int DEFAULT_GROUPBY_TRIM_THRESHOLD = 1_000_000;
+  @Deprecated
```

Comment:
It'd be good to add a comment explaining why these are deprected and what should be used instead.

Reply:
Done